lldb-eval/api.h

```diff
@@ -73,6 +98,43 @@ LLDB_EVAL_API
 lldb::SBValue EvaluateExpression(lldb::SBValue scope, const char* expression,
                                  Options opts, lldb::SBError& error);
 
+LLDB_EVAL_API
+ParsedExpression ParseExpression(lldb::SBFrame, const char* expression,
+                                 lldb::SBError& error);
+
+LLDB_EVAL_API
+ParsedExpression ParseExpression(lldb::SBFrame, const char* expression,
+                                 Options opts, lldb::SBError& error);
+
+LLDB_EVAL_API
+ParsedExpression ParseExpression(lldb::SBTarget target, lldb::SBType scope,
+                                 const char* expression, lldb::SBError& error);
+
+LLDB_EVAL_API
+ParsedExpression ParseExpression(lldb::SBTarget target, lldb::SBType scope,
```

Comment:
Let's call it `CompileExpression`, we're getting closer and closer to mini-compiler here :)
The return value can be `CompiledExpr` for shortness.